contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -280,112 +180,107 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
+            revert Errors.IpRoyaltyVault__NotWhitelistedRoyaltyToken();
+        if (amount == 0) revert Errors.IpRoyaltyVault__ZeroAmount();
 
-    /// @notice Amount of revenue token claimable at a given snapshot
-    /// @param snapshotId The snapshot id
-    /// @param token The address of the revenue token
-    function claimableAtSnapshot(uint256 snapshotId, address token) external view returns (uint256) {
-        return _getIpRoyaltyVaultStorage().claimableAtSnapshot[snapshotId][token];
-    }
+        $.tokens.add(token);
+        $.poolInfo[token] += amount;
 
-    /// @notice Indicates whether the claimer has claimed the revenue tokens at a given snapshot
-    /// @param snapshotId The snapshot id
-    /// @param claimer The address of the claimer
-    /// @param token The address of the revenue token
-    function isClaimedAtSnapshot(uint256 snapshotId, address claimer, address token) external view returns (bool) {
-        return _getIpRoyaltyVaultStorage().isClaimedAtSnapshot[snapshotId][claimer][token];
+        emit RevenueTokenAddedToVault(token, amount);
     }
 
-    /// @notice Returns list of revenue tokens in the vault
-    function tokens() external view returns (address[] memory) {
-        return (_getIpRoyaltyVaultStorage().tokens).values();
-    }
+    function _update(address from, address to, uint256 amount) internal override {
+        if (from == address(0)) {
+            super._update(from, to, amount);
+            return;
+        }
+        if (from == to) revert Errors.IpRoyaltyVault__SameFromToAddress(address(this), from);
+        // when transferring RoyaltyTokens (Vault) from a user to another user:
+        // 1. clear pending rewards of the (from) user
+        // 2. clear pending rewards of the (to) another user, if pending rewards of another user is not zero
+        // 3. update the rewardDebt of the (to) another user to accBalancePerShare * userAmount
+        // 4. update the rewardDebt of the (from) user to accBalancePerShare * userAmount
+        // 5. transfer the RoyaltyTokens (Vault) from the (from) user to the (to) another user
+        if (balanceOf(from) == 0) revert Errors.IpRoyaltyVault__ZeroBalance(address(this), from);
+        if (balanceOf(from) < amount) revert Errors.IpRoyaltyVault__InsufficientBalance(address(this), from, amount);
 
-    /// @notice A function to calculate the amount of revenue token claimable by a token holder at certain snapshot
-    /// @param account The address of the token holder
-    /// @param snapshotId The snapshot id
-    /// @param token The revenue token to claim
-    /// @return The amount of revenue token claimable
-    function _claimableRevenue(address account, uint256 snapshotId, address token) internal view returns (uint256) {
         IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
+        address[] memory tokenList = $.tokens.values();
+        uint256 totalSupply = totalSupply();
```

Comment:
`uint256 totalSupply = totalSupply();` reads state and is a step where we could save on gas by replacing with an immutable variable of the vault

Something along the lines of:

function totalSupply() public view override returns (uint256) {
        return TOTAL_SUPPLY;
   }

And then can eliminate `uint256 totalSupply = totalSupply();` and use `TOTAL_SUPPLY` directly

Given all vaults have the same total supply and it stays constant over the lifetime of the vault (no minting and no burning) - this is possible to do to save gas on every royalty token transfer.

Reply:
Good point!
created an issue to track this enhancement: #294. The change will not be trivial, as it requires modifications to the RoyaltyModule to ensure they share the same configuration. Additionally, updates to the DeployHelper may also be necessary. Therefore, it would be better to handle this in a separate PR.